src/Bridge/Graphql/Resolver/ItemMutationResolverFactory.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Graphql\Resolver;
+
+use ApiPlatform\Core\Api\IdentifiersExtractorInterface;
+use ApiPlatform\Core\Api\UrlGeneratorInterface;
+use ApiPlatform\Core\Bridge\Symfony\Routing\RouteNameGenerator;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use GraphQL\Error\Error;
+use GraphQL\Type\Definition\ResolveInfo;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+/**
+ * Creates a function resolving a GraphQL mutation of an item.
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ *
+ * @internal
+ */
+final class ItemMutationResolverFactory implements ItemMutationResolverFactoryInterface
+{
+    private $httpKernel;
+    private $urlGenerator;
+    private $resourceMetadataFactory;
+    private $identifiersExtractor;
+
+    public function __construct(HttpKernelInterface $httpKernel, UrlGeneratorInterface $urlGenerator, ResourceMetadataFactoryInterface $resourceMetadataFactory, IdentifiersExtractorInterface $identifiersExtractor)
+    {
+        $this->httpKernel = $httpKernel;
+        $this->urlGenerator = $urlGenerator;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->identifiersExtractor = $identifiersExtractor;
+    }
+
+    public function createItemMutationResolver(string $resourceClass, string $operationName, array $operation): callable
+    {
+        return function ($root, $args, $context, ResolveInfo $info) use ($resourceClass, $operationName, $operation) {
+            $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
+            $resourceShortName = $resourceMetadata->getShortName();
+
+            $route = RouteNameGenerator::generate($operationName, $resourceShortName, 'item');
+            $identifiers = $this->identifiersExtractor->getIdentifiersFromResourceClass($resourceClass);
+            if (count($identifiers) > 1) {
+                $identifierPairs = array_map(function ($identifier) use ($args) {
+                    return "{$identifier}={$args['input'][$identifier]}";
+                }, $identifiers);
+                $id = implode(';', $identifierPairs);
+            } else {
+                $id = $args['input'][$identifiers[0]];
+            }
+            $uri = $this->urlGenerator->generate($route, ['id' => $id]);
+            $request = Request::create($uri, $operation['method']);
+            $request->headers->set('Accept', 'application/json');
+
+            $response = $this->httpKernel->handle($request, HttpKernelInterface::SUB_REQUEST);
```

Comment:
This is weird but I get it, we don't actually have anything to handle object deletion, we just call `remove()` from Doctrine in the Doctrine's `WriteListener`: https://github.com/api-platform/core/blob/master/src/Bridge/Doctrine/EventListener/WriteListener.php#L62

It was OK for REST until now but it doesn't scale to GraphQL. Injecting `HttpKernel` here is bad design.

I suggest to do the following:

* Introduce a new `ObjectManagerInterface` in API Platform with `delete()` and a `persist()` method
* Implement this interface in the Doctrine bridge to delegate to Doctrine's Object Manager (it will just be a proxy class)
* Use this new interface here and in `WriteListener`

WDYT?

Reply:
Ok so you want to modify API Platform to avoid this issue then. Your suggestion seems good to me.

Reply:
Yes, it's a tiny, non-breaking change IMO.

Reply:
The existing `Bridge\Doctrine\EventListener\WriteListener` should be deprecated (and not registered anymore) and a new one (`EventListener\WriteListener`) using the new interface should be added.

Reply:
Can you please do this in another PR?

Reply:
Why not but this other PR must be merged before this one :)

Reply:
Obsly :)